cds/base/dojson/marc21/__init__.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of CERN Document Server.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+from invenio.utils.datastructures import SmartDict
+
+from invenio_search.api import Query
+
+from .registry import translations as _translations
+
+
+def convert_cdsmarcxml(source):
+    """Convert CDS to JSON."""
+    from dojson.contrib.marc21.utils import create_record, split_blob
+
+    for data in split_blob(source.read()):
+        record = create_record(data)
```

Comment:
We need to be careful here, `create_record` returns a `SaveDict` object:
```python
class SaveDict(dict):
    __getitem__ = dict.get
```
But we need a `SmartDict` object to use the dot syntax in que search query, i.e. `690C_.a:PHOTO`.

Maybe something like this should solve the issue (probably there other ways):
```python
from invenio.utils.datastructures import SmartDict
record = SmartDict(dict(create_record(xml)))
```
Example:
```ipython
In [1]: from invenio_search.api import Query

In [2]: from invenio.utils.datastructures import SmartDict

In [3]: from dojson.contrib.marc21.utils import create_record

In [4]: xml = open('photo.xml', 'r').read()

In [5]: r = SmartDict(dict(create_record(xml)))

In [6]: q = Query('690C_.a:PHOTO')

In [7]: q.match(r)
Out[7]: True
```

I think we should also add a simple test case to verify that the `Query` is working as we suppose it to.